tests/chart_tests/test_sts_volume_claim_annotations.py

```diff
@@ -0,0 +1,30 @@
+import pytest
+
+from tests import supported_k8s_versions
+from tests.chart_tests.helm_template_generator import render_chart
+
+
+@pytest.mark.parametrize(
+    "kube_version",
+    supported_k8s_versions,
+)
+class TestElasticSearch:
+    def test_elasticsearch_with_default_annotations(self, kube_version):
+        """Test all sts for the volume claim templates annotations"""
+        docs = render_chart(
+            kube_version=kube_version,
+            values={},
+            show_only=[
+                "charts/elasticsearch/templates/master/es-master-statefulset.yaml",
+                "charts/elasticsearch/templates/data/es-data-statefulset.yaml",
+                "charts/alertmanager/templates/alertmanager-statefulset.yaml",
+                "charts/astronomer/templates/registry/registry-statefulset.yaml",
+                "charts/prometheus/templates/prometheus-statefulset.yaml",
+                "charts/stan/templates/statefulset.yaml",
+            ],
+        )
+
+        assert len(docs) == 6
+        for doc in docs:
+            if "annotations" in doc["spec"]["volumeClaimTemplates"][0]["metadata"]:
+                assert doc["spec"]["volumeClaimTemplates"][0]["metadata"]["annotations"]
```

Comment:
We should set `.Values.persistence.annotations` on line 16 and make some assertions about them here

Reply:
this case if for default and we need one more test case for overrides with custom annotations added

Reply:
If annotations are there, we should assert something. If they are not there, we should assert that they are not there.